Framework/DataHandling/src/LoadInstrument.cpp

```diff
@@ -198,62 +198,71 @@ void LoadInstrument::idfInstrumentLoader() {
       // Add to data service for later retrieval
       InstrumentDataService::Instance().add(instrumentNameMangled, instrument);
     }
-    m_workspace->setInstrument(instrument);
+    ws->setInstrument(instrument);
 
     // populate parameter map of workspace
-    m_workspace->populateInstrumentParameters();
+    ws->populateInstrumentParameters();
 
     // LoadParameterFile modifies the base instrument stored in the IDS so this
     // must also be protected by the lock until LoadParameterFile is fixed.
     // check if default parameter file is also present, unless loading from
-    if (!m_filename.empty())
-      runLoadParameterFile();
+    if (!filename.empty())
+      runLoadParameterFile(ws, filename);
   }
 }
 
-void LoadInstrument::nexusInstrumentLoader() {
-  checkAndRetrieveInstrumentFilename(FileType::Nexus);
+void LoadInstrument::nexusInstrumentLoader(boost::shared_ptr<API::MatrixWorkspace>& ws,
```

Comment:
As you have eliminated member variables, you could consider making this method `const` now or even a non-member.  As a non-member this would enable you to keep the declaration out of the header. Better for compile times and long-term maintenance.